guava/src/com/google/common/collect/Maps.java

```diff
@@ -523,6 +524,38 @@ MapDifference<K, V> difference(
     return new MapDifferenceImpl<>(onlyOnLeft, onlyOnRight, onBoth, differences);
   }
 
+  /**
+   * Computes the difference between two maps. This difference is an immutable snapshot of the state
+   * of the maps at the time this method is called. It will never change, even if the maps change at
+   * a later time.
+   *
+   * <p>Since this method uses {@code HashMap} instances internally, the keys of the supplied maps
+   * must be well-behaved with respect to {@link Object#equals} and {@link Object#hashCode}.
+   *
+   * @param left the map to treat as the "left" map for purposes of comparison
+   * @param right the map to treat as the "right" map for purposes of comparison
+   * @param equivalence the equivalence relationship to use to compare values
+   * @return the difference between the two maps
+   * @since 10.0
+   */
+  public static <K extends @Nullable Object, V extends @Nullable Object>
+      MapDifference<K, V> difference(
+          Map<? extends K, ? extends V> left,
+          Map<? extends K, ? extends V> right,
+          BiPredicate<? super @NonNull V, ? super @NonNull V> equivalence) {
+    Preconditions.checkNotNull(equivalence);
+    return difference(left, right, new Equivalence<V>() {
+      @Override
+      protected boolean doEquivalent(V a, V b) {
+        return equivalence.test(a, b);
+      }
+      @Override
+      protected int doHash(V v) {
+        return v.hashCode();
+      }
+    });
```

Comment:
Instead of creating this custom `Equivalence` just for `Maps#difference`, would it be an idea to instead introduce a new static factory method `Equivalence#equals(BiPredicate)`? That would allow the new feature to be combined with all other `Equivalence`-accepting APIs. (Downside may be that this alternative is less "discoverable" :thinking:.)

Reply:
Good idea! I also think the static factory is better, because this is just a lightly used method. I will change the code and pull request again.